modules/rest/src/main/java/org/apache/ignite/internal/rest/RestComponent.java

```diff
@@ -51,6 +53,11 @@
  * resources for the example.
  */
 public class RestComponent implements IgniteComponent {
+    /**
+     * Lock for micronaut server startup.
+     * TODO: remove when fix https://github.com/micronaut-projects/micronaut-core/issues/10091
+     */
+    private static final Lock SHARED_STARTUP_LOCK = new ReentrantLock();
```

Comment:
Just curious: why do you use a `Lock` and not `synchronized` on a static Object?